django/contrib/admin/options.py

```diff
@@ -109,9 +109,8 @@ class BaseModelAdmin(six.with_metaclass(forms.MediaDefiningClass)):
     show_full_result_count = True
     checks_class = BaseModelAdminChecks
 
-    @classmethod
-    def check(cls, model, **kwargs):
-        return cls.checks_class().check(cls, model, **kwargs)
+    def check(obj, model, **kwargs):
+        return obj.checks_class().check(obj, model, **kwargs)
```

Comment:
Actually this should be `classmethod` for `InlineModelAdmin` and a regular method for `ModelAdmin`.